File: 프로그래머스/lv1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.py
def solution(n, m):
    answer = [0, 0]
    if n > m:
        n, m = m, n 
    
    # 최대공약수
    for i in range(m, 0, -1):      # 최대 공약수: n < m일 때 1 <= x <= m
        if n % i == 0 and m % i == 0:
            answer[0] = i
            break
    
    # 최소공배수
    for j in range(m, n*m+1):       # 최소 공배수: n < m일 때, m <= y <= n*m
        if j % n == 0 and j % m == 0:
            answer[1] = j
            break
        
    return answer

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
class Solution {
    public int[] solution(int n, int m) {
        int[] answer = {1, 1};
        
        // 최대공약수
        for (int i = 2; i < n * m; i++) {
            while ((n % i == 0) && (m % i == 0)) {
                n /= i;
                m /= i;
                answer[0] *= i;
            }
        }
        
        // 최소공배수 = 최대공약수 * n의 나머지 * m의 나머지        
        answer[1] = answer[0] * n * m;
        
        
        return answer;
    }
}